### PR TITLE
Add generic vision simulation implementation

### PR DESCRIPTION
**Pull request by virtuald**
This seems to work well. The coordinates I'm using for the 2017 game are as follows:

```
targets = [
    # right
    VisionSim.Target(16, 12, 250, 20), # angle is 122.23
    # middle
    VisionSim.Target(18.5, 16, 295, 65), # angle is 180
    # left
    VisionSim.Target(16, 20, 340, 110), # angle is -142
]
```

### CODE REVIEW

File: lib/pyfrc/physics/visionsim.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+                                  self.view_angle_start,
+                                  self.view_angle_end):
+            return
+        
+        # is the robot facing the target?
+        a = angle - self.fov2
+        b = angle + self.fov2
+        
+        if _in_angle_interval(target_angle + math.pi, a + math.pi, b + math.pi):
+            offset = math.degrees((((target_angle - angle) + math.pi) % (math.pi*2)) - math.pi)
```
Comment (auscompgeek) [inline lib/pyfrc/physics/visionsim.py]: Probably a good idea to use `math.atan2(math.sin(angle), math.cos(angle))` rather than float modulo.

Comment (virtuald) [inline lib/pyfrc/physics/visionsim.py]: breaks the tests.

Comment (auscompgeek) [inline lib/pyfrc/physics/visionsim.py]: Really? Hmm.

**Comment by virtuald** [comment]
@auscompgeek I have to head to bed, but I would love to wake up and find a PR from you that improves the math on this to be more efficient. 👍 

I'll push a new version of pyfrc in the morning either way.

File: lib/pyfrc/physics/visionsim.py (hunk 1 above)
Comment (auscompgeek) [inline lib/pyfrc/physics/visionsim.py]:
```python
offset = math.degrees(math.atan2(math.sin(target_angle - angle), math.cos(target_angle - angle)))
```
worked fine for me.

**Pull request merged**